integration_test.go

```diff
@@ -80,8 +84,174 @@ func BenchmarkStartup(b *testing.B) {
+				http.MethodGet: {
+					PermissionV2: &core.RondConfig{
+						RequestFlow: core.RequestFlow{PolicyName: "filter_something", GenerateQuery: true},
+					},
+				},
+			},
+			"/project-data": {
+				http.MethodPost: {
+					PermissionV2: &core.RondConfig{
+						RequestFlow:  core.RequestFlow{PolicyName: "allow_post"},
+						ResponseFlow: core.ResponseFlow{PolicyName: "proj_data"},
+					},
+				},
+			},
+		},
+	}
+	oasFileName := writeOAS(t, tmpdir, oas)
+	policiesFileName := writePolicies(t, tmpdir, policies)
+
+	defer gock.Off()
+	defer gock.DisableNetworkingFilters()
+	defer gock.DisableNetworking()
+
+	gock.EnableNetworking()
+	gock.NetworkingFilter(func(r *http.Request) bool {
+		if r.URL.Host == "localhost:3050" {
+			return false
+		}
+		if r.URL.Path == "/documentation/json" && r.URL.Host == "localhost:3050" {
+			return false
+		}
+		return true
+	})
+
+	gock.New("http://localhost:3050").
+		Persist().
+		Get("/documentation/json").
+		Reply(200).
+		File(oasFileName)
+
+	gock.New("http://localhost:3050/").
+		Persist().
+		Get("/allow-get").
+		Reply(200)
+	gock.New("http://localhost:3050/").
+		Persist().
+		Get("/filter-something").
+		Reply(200)
+	gock.New("http://localhost:3050/").
+		Persist().
+		Post("/project-data").
+		Reply(200).
+		JSON([]string{})
+
+	mongoHost := os.Getenv("MONGO_HOST_CI")
+	if mongoHost == "" {
+		mongoHost = testutils.LocalhostMongoDB
+		t.Logf("Connection to localhost MongoDB, on CI env this is a problem!")
+	}
+	randomizedDBNamePart := testutils.GetRandomName(10)
+	mongoDBName := fmt.Sprintf("test-%s", randomizedDBNamePart)
+	envs, err := env.ParseAsWithOptions[config.EnvironmentVariables](env.Options{
+		Environment: map[string]string{
+			"TARGET_SERVICE_HOST":      "localhost:3050",
+			"TARGET_SERVICE_OAS_PATH":  "/documentation/json",
+			"OPA_MODULES_DIRECTORY":    policiesFileName,
+			"LOG_LEVEL":                "fatal",
+			"MONGODB_URL":              fmt.Sprintf("mongodb://%s/%s", mongoHost, mongoDBName),
+			"BINDINGS_COLLECTION_NAME": "bindings",
+			"ROLES_COLLECTION_NAME":    "roles",
+		},
+	})
+	require.NoError(t, err)
+
+	app, err := setupService(envs, log)
+	require.NoError(t, err)
+	require.True(t, <-app.sdkBootState.IsReadyChan())
+	defer app.close()
+
+	// everything is up and running, now start bombarding the webserver
+	type RequestConf struct {
+		Verb           string
+		Path           string
+		ExpectedStatus int
+	}
+	dictionary := []RequestConf{
+		{Verb: http.MethodGet, Path: "/allow-get", ExpectedStatus: http.StatusOK},
+		{Verb: http.MethodPost, Path: "/allow-get", ExpectedStatus: http.StatusForbidden},
+		{Verb: http.MethodGet, Path: "/filter-something", ExpectedStatus: http.StatusOK},
+		{Verb: http.MethodPost, Path: "/filter-something", ExpectedStatus: http.StatusNotFound},
+		{Verb: http.MethodPost, Path: "/project-data", ExpectedStatus: http.StatusOK},
+		{Verb: http.MethodGet, Path: "/project-data", ExpectedStatus: http.StatusNotFound},
```

Comment:
Also here, can be interesting to assert that data is correctly projected?

Reply:
done